macros/snowflake_snowpark/create_stored_procedure.sql

```diff
@@ -0,0 +1,514 @@
+{% macro create_report_table_proc() %}
+  {{ return(adapter.dispatch('create_report_table_proc', 'snowplow_fractribution')()) }}
+{% endmacro %}
+
+{% macro default__create_report_table_proc() %}
+{% endmacro %}
+
+{% macro snowflake__create_report_table_proc() %}
+{% set stored_proc %}
+create or replace procedure {{target.schema}}_derived.create_report_table(attribution_model STRING, conversion_window_start_date STRING, conversion_window_end_date STRING)
```

Comment:
How do I get this to create the procedure in `_derived` without hardcoding it...?

Reply:
I'd have to double check this, but I _think_ the `target.schema` is relative to the model that called the macro itself, however if it's called via a hook then I think it's just the default profile target so you'd have to hardcode it...

Reply:
Okay so i've figured out that I should actually be using `{{schema}}` not `{{target.schema}}` as the former compiles to the custom schema without any hardcoding. And it seems to work whether it's in the on-run-end in dbt_project.yml or a post_hook in a model config.